<!doctype html>
<html class="no-js">
<head>
    <meta charset="utf-8">
    <meta name="description" content="Innovation and Open Source">
    <meta name="author" content="">
    <title>TupiLabs</title>
    <meta name="description" content="">
    <meta name=viewport content="width=device-width, initial-scale=1">
    <meta name="mobile-web-app-capable" content="yes">
    
    <link rel="stylesheet" type="text/css" href="/semantic.min.css">
    <link rel="stylesheet" type="text/css" href="/site.css">
</head>
<body>
    <!-- Following Menu -->
    <div class="ui large top fixed menu">
        <div class="ui container">
            <a class="item" href="/"><i class="home icon"></i> Home</a>
            <a class="item" href="/work"><i class="terminal icon"></i> Work</a>
            <a class="item" href="/tutorials"><i class="book icon"></i> Tutorials</a>
            <a class="item" href="/blog"><i class="newspaper icon"></i> Blog</a>
            <a class="item" href="/courses"><i class="student icon"></i> Courses</a>
            <a class="item" href="/lab"><i class="lab icon"></i> Lab</a>
            <div class="right menu">
                <a class="item" href="/contact"><i class="mail outline icon"></i> Contact</a>
            </div>
        </div>
    </div>
    
    <!-- Page Contents -->
    <div class="pusher">
        <div class="ui container">
            <div class="ui large stackable menu">
                <a class="item" href="/"><i class="home icon"></i> Home</a>
                <a class="item" href="/work"><i class="terminal icon"></i> Work</a>
                <a class="item" href="/tutorials"><i class="book icon"></i> Tutorials</a>
                <a class="item" href="/blog"><i class="newspaper icon"></i> Blog</a>
                <a class="item" href="/courses"><i class="student icon"></i> Courses</a>
                <a class="item" href="/lab"><i class="lab icon"></i> Lab</a>
                <div class="right menu">
                    <a class="item" href="/contact"><i class="mail outline icon"></i> Contact</a>
                </div>
            </div>
        </div>
        
        
        

<div class="ui basic segment" id="blog">
    <div class="ui stackable grid container">
        <div class="row">
            <div class="sixteen wide column">
                <h1>Posts tagged with bioinformatics</h1>
            </div>
        </div>
        <div class="row">
            <div class="sixteen wide column">
                <section>
                    
                    <div class="ui padded segment">
  <article>
    <header>
      <h2><a href="/2013/05/03/enhance-mysql-management-in-stacks-university-of-oregon">Enhance MySQL management in Stacks (University of Oregon)</a></h2>
      <small>May 03, 2013
        
        in 
            <a href="/tag/ideas">ideas</a>, <a href="/tag/bioinformatics">bioinformatics</a>
        
        
        | <a href="/blog">blog</a>
        
      </small>
    </header>
    <section>
      <p>
 From <a href="https://groups.google.com/forum/?fromgroups=#!topic/stacks-users/2HwY2dDyeSM">https://groups.google.com/forum/?fromgroups=#!topic/stacks-users/2HwY2dDyeSM</a>.
</p>

<p>Stacks processes RAD DNA sequences and produces output, that is displayed on the Web with a bundled PHP+MySQL
 application. This application is quite useful, especially since it helps developers to analyze large amounts of data.</p>

<p>
 <a href="http://creskolab.uoregon.edu/stacks/"><img
  src="img/logos/stacks.png" alt="stacks_logo" width="300" height="156"
  class="aligncenter size-medium wp-image-1256" /></a>
</p>

<p>One limitation though, is that some management tasks may need manual execution of SQL&#8217;s in the database. Someone
 could add few scripts or new features to the Web interface (like a management section?). This could help researchers on
 their research.</p>
    </section>
  </article>
</div>
                    
                    <div class="ui padded segment">
  <article>
    <header>
      <h2><a href="/2013/03/26/work-on-biojava-and-biosql-integration">Work on BioJava and BioSQL integration</a></h2>
      <small>Mar 26, 2013
        
        in 
            <a href="/tag/bioinformatics">bioinformatics</a>, <a href="/tag/ideas">ideas</a>
        
        
        | <a href="/blog">blog</a>
        
      </small>
    </header>
    <section>
      <p><a href="http://www.biojava.org" title="BioJava">BioJava</a> code has two versions, the current one (version 3) and the legacy code. In one of these month&#8217;s messages to the <a href="http://lists.open-bio.org/pipermail/biojava-dev/2013-March/005114.html">BioJava development mailing list</a>: </p>

<p><quote>&#8220;Unfortunately, biojava 3 does not have any support for biosql at this
stage. If you want to use that, you will have to use the biojava 1.x
series&#8230;&#8221;</quote></p>
<p>BioJava is extremely useful for Java projects, researchers, libraries and tools. And the same can be said about as <a href="http://www.biosql.org">BioSQL</a>. The integration of these two tools would bring many benefits to the whole community (private companies, researchers, institutions, universities, etc).</p>
    </section>
  </article>
</div>
                    
                    <div class="ui padded segment">
  <article>
    <header>
      <h2><a href="/2013/02/17/tupilabs-report-feb-10-feb-16">Tupilabs Report: Feb 10, Feb 16</a></h2>
      <small>Feb 17, 2013
        
        in 
            <a href="/tag/bioinformatics">bioinformatics</a>, <a href="/tag/biotechnology">biotechnology</a>, <a href="/tag/biouno">biouno</a>, <a href="/tag/jenkins-en">jenkins-en</a>, <a href="/tag/speak-like-a-brazilian">speak-like-a-brazilian</a>, <a href="/tag/tupilabs-report">tupilabs-report</a>
        
        
        | <a href="/news">news</a>
        
      </small>
    </header>
    <section>
      <p>Hereâ€™s the list of the cool things we did since last Sunday at TupiLabs.</p>

<ul>
    <li>Migrated <a href="https://github.com/tupilabs/cvb">CVB</a> from SourceForge.net to GitHub, to start over the development</li>
    <li>Created <a href="https://github.com/tupilabs/structure-harvester-plugin">Harvester Plug-in</a> project for BioUno</li>
    <li>Created <a href="https://github.com/tupilabs/clumpp-plugin">Clumpp Plug-in</a> project for BioUno</li>
    <li>Added <a href="http://en.wikipedia.org/wiki/Defuzzification">defuzzification functions</a> to <a href="http://nebular.sf.net">nebular</a></li>
    <li>Released <a href="http://nebular.sf.net">nebular 0.3</a> to Maven central</li>
    <li>Migrated <a href="https://github.com/tupilabs/bellmanzadeh">BellmanZadeh</a> from SourceForge.net to GitHub, to start over the development</li>
    <li>Started a new prototype project with a prospect</li>
    <li>New expressions in <a href="http://www.speaklikeabrazilian.com">Speak Like A Brazilian</a></li>
</ul>

<p>We are working <a href="for_you/" title="For you">for you</a></p>

<p>Have a great week! :D</p>
    </section>
  </article>
</div>
                    
                    <div class="ui padded segment">
  <article>
    <header>
      <h2><a href="/2013/02/05/study-if-its-doable-to-parallelize-pacbiotoca">Study if it&#8217;s doable to parallelize PacBioToCA</a></h2>
      <small>Feb 05, 2013
        
        in 
            <a href="/tag/bioinformatics">bioinformatics</a>, <a href="/tag/ideas">ideas</a>
        
        
        | <a href="/blog">blog</a>
        
      </small>
    </header>
    <section>
      <p><a href="http://www.homolog.us/">Homolog.us</a> is an amazing site. The content is clear, rich and very entertaining (at least for those who like biology and technology). They have an active <a href="https://twitter.com/#!/homolog_us/">Twitter account</a>, they answer comments, they are on <a href="http://reddit.com/r/bioinformatics">reddit</a> too. <strong>These guys rock</strong>!</p>

<p>Recently <a href="http://www.homolog.us/blogs/2013/02/05/pacbiotoca-for-error-correcting-pacbio-reads/">they wrote</a> about <a href="http://sourceforge.net/apps/mediawiki/wgs-assembler/index.php?title=PacBioToCA">PacBioToCA</a>, pointing out its lengthy execution. It&#8217;s common in biotechnology parallel execution with <a href="http://en.wikipedia.org/wiki/Open_MPI" title="">OpenMPI</a>, <a href="http://en.wikipedia.org/wiki/OpenMP">OpenMP</a> or other method like <a href="http://en.wikipedia.org/wiki/MapReduce">MapReduce</a>.</p>

<p>One idea would be review the code, check if it&#8217;s doable some kind of parallelism and write about it. Perhaps rewriting it. They <a href="http://www.homolog.us/blogs/2013/02/05/lsc-an-alternative-to-pacbiotoca-for-error-correction-of-pacbio-reads/">already wrote</a> about another tool that competes with PacBioToCA. I told you that Homolog.us is amazing.</p>
    </section>
  </article>
</div>
                    
                    <div class="ui padded segment">
  <article>
    <header>
      <h2><a href="/2012/11/06/running-stacks-analysis-on-a-supercomputer-no-mysql-and-loading-the-results-into-an-existing-database">Running Stacks denovo pipeline on a supercomputer (no MySQL) and loading the results into an existing database</a></h2>
      <small>Nov 06, 2012
        
        in 
            <a href="/tag/bioinformatics">bioinformatics</a>
        
        
        | <a href="/tutorials">tutorials</a>
        
      </small>
    </header>
    <section>
      <p>
 In this post you will see a short description of one method to run the <a
  href="http://www.g3journal.org/content/1/3/171.full"
  title="Stacks: Building and Genotyping Loci De Novo From Short-Read Sequences">denovo</a> pipeline with <a
  href="http://creskolab.uoregon.edu/stacks/" title="Stacks">Stacks</a> on a supercomputer that has no MySQL database,
 and how to load the results later in another computer with the database installed.
</p>

<blockquote>Stacks is a software pipeline for building loci out of a set of short-read sequenced samples.
 Stacks was developed for the purpose of building genetic maps from RAD-Tag Illumina sequence data, but can also be
 readily applied to population studies, and phylogeography.</blockquote>

<p>
 In order to run Stacks in parallel you need <a href="http://openmp.org/wp/" title="OpenMP">OpenMP</a>. What also means
 that if you need computer power, probably you will want a good amount of memory and many CPU cores. A common case is
 running Stacks in supercomputers, where you can get nodes with 8, 16 or more CPU&#8217;s.
</p>

<p class='center'>
 <a href="http://creskolab.uoregon.edu/stacks/"><img src="img/logos/stacks.png"
  alt="" title="Stacks" width="300" height="156" class="alignnone size-medium wp-image-941" /></a>
</p>

<p>
 However, it is not normal to have databases in these servers. Usually the job scheduling is controlled via a batch
 server (like <a href="http://en.wikipedia.org/wiki/Portable_Batch_System" title="PBS">PBS</a>) or some other computer
 facility management tool. Having a database in the same server, the CPU&#8217;s would have to handle hundreds or thousands of
 connections, what could interfere with the job scheduling
</p>

<p>
 The denovo analysis in Stacks is achieved through the <em>denovo_map.pl</em> Perl script. What this script does,
 basically, is read a bunch of files (samples) and the output of other programs, call other programs (like <em>ustacks</em>,
 <em>sstacks</em>, <em>genotypes</em>, etc) and load data into a MySQL database.
</p>

<p>
 The data is load incrementally as the programs are executed. You can disable the MySQL interaction by using the <em>-S</em>
 flag. But probably you&#8217;ll want to load it later. Stacks comes with a very useful web application (written in PHP +
 MDB2). This web application scans your MySQL server looking for databases like %radtags, and produces a nice interface
 for mining data. You can sort, browse and filter your sequences, look at SNPs and other useful data.
</p>

<p class='center'>
 <a href="img/posts/2012-11-06_running-stacks-analysis-on-a-supercomputer-no-mysql-and-loading-the-results-into-an-existing-database/stacks_screenshot.png"><img
  src="img/posts/2012-11-06_running-stacks-analysis-on-a-supercomputer-no-mysql-and-loading-the-results-into-an-existing-database/stacks_screenshot.png" alt="" title="Stacks web application screen shot"
  class="alignnone size-medium wp-image-943" /></a>
</p>

<p>
 There is an easy way to load your data into a MySQL database after your analysis finished in the supercomputer (and you
 have used the -S flag). You can use <em>load_radtags.pl</em>. Here&#8217;s a short summary of the actions you would have to
 follow.
</p>

<ul>
 <li>Download your samples directory, and the directory where the program saved tags, snps, matches and a lot of
  other tsv files (including denovo_map.log).</li>
 <li>Execute load_radtags.pl using the right parameters</li>
</ul>

<p>
 You can also use the <em>-d</em> option (dry run). This options prints what is going to be executed, but doesn&#8217;t run
 your analysis. It&#8217;s good to use this option when you are not 100% sure about your command. Below is an example of how
 to use load_radtags.pl. It also includes a call to index_radtags.pl, as you have to call it in order to be able to see
 your data in the web interface.
</p>

<p><code>$ load_radtags.pl -D pe_radtags -p stacks -b 1 -c -B -e "PE populations RAD" -t population $
 index_radtags.pl -D pe_radtags -c -t</code></p>
<p>If you used the right parameters, and didn&#8217;t see errors in the console, your data should be available in the web
 interface now. Easy, no? :-)</p>
    </section>
  </article>
</div>
                    
                </section>
                <hr class="ui divider" />
                <div class="ui pagination menu">
                    
                    <a class="item" href="/tag/bioinformatics/2"><i class='left arrow icon'></i> Previous Posts</a>
                    
                    
                </div>
            </div>
        </div>
    </div>
</div>
        
        
        
<div class="ui inverted vertical footer segment">
  <div class="ui container">
    <div class="ui stackable inverted divided equal height stackable grid">
      <div class="three wide column">
        <h4 class="ui inverted header">Resources</h4>
        <div class="ui inverted link list">
          <a href="/about" class="item">About</a>
          <a href="/contact" class="item">Contact Us</a>
        </div>
      </div>
      <div class="three wide column">
        <h4 class="ui inverted header">Follow Us</h4>
        <div class="ui inverted link list">
          <a href="https://github.com/tupilabs" class="item"><i class="github icon"></i> GitHub</a>
          <a href="https://www.twitter.com/tupilabs" class="item"><i class="twitter icon"></i> Twitter</a>
          <a href="https://www.facebook.com/pages/TupiLabs/367986299948160" class="item"><i class="facebook icon"></i> FaceBook</a>
          <a href="http://www.youtube.com/user/TupiLabs" class="item"><i class="youtube icon"></i> YouTube</a>
        </div>
      </div>
      <div class="seven wide column">
        <h4 class="ui inverted header">Check out our lab!</h4>
        <p>Before projects are incubated in TupiLabs, we implement small proof of concepts in our
          <a href="/lab">lab</a>.
          You can check out what we are working on, and help us to make it better by telling us what
          you think.</p>
      </div>
    </div>
  </div>
</div>
    </div>
    <script src="/jquery.min.js"></script>
    <script src="/semantic.min.js"></script>
    <script src="/site.js"></script>
    <script>
    (function(i,s,o,g,r,a,m){i['GoogleAnalyticsObject']=r;i[r]=i[r]||function(){
        (i[r].q=i[r].q||[]).push(arguments)},i[r].l=1*new Date();a=s.createElement(o),
        m=s.getElementsByTagName(o)[0];a.async=1;a.src=g;m.parentNode.insertBefore(a,m)
    })(window,document,'script','//www.google-analytics.com/analytics.js','ga');
    
    ga('create', 'UA-27590622-1', 'auto');
    ga('send', 'pageview');
    
    </script>
</body>
</html>